plugins/out_stackdriver/stackdriver.c

```diff
@@ -624,20 +674,37 @@ static int stackdriver_format(const void *data, size_t bytes,
             msgpack_pack_int(&mp_pck, severity);
         }
 
+        /* Add insertId field into the log entry */
+        if (insertId_extracted) {
+            msgpack_pack_str(&mp_pck, 8);
+            msgpack_pack_str_body(&mp_pck, "insertId", 8);
+            msgpack_pack_object(&mp_pck, insertId_obj);
+        }
+
         /* Add operation field into the log entry */
         if (operation_extracted) {
             add_operation_field(&operation_id, &operation_producer,
                                 &operation_first, &operation_last, &mp_pck);
         }
+
+        /* Add sourceLocation field into the log entry */
+        if (sourceLocation_extracted) {
+            add_sourceLocation_field(&sourceLocation_file, sourceLocation_line, 
+                                &sourceLocation_function, &mp_pck);
+        }
         
-        /* Clean up id and producer if operation extracted */
+        /* Clean up */
+        flb_sds_destroy(insertId_key);
         flb_sds_destroy(operation_id);
         flb_sds_destroy(operation_producer);
+        flb_sds_destroy(sourceLocation_file);
+        flb_sds_destroy(sourceLocation_function);
 
```

Comment:
if you rely on the default value of `sourceLocation_line`, you need to assign the value of `sourceLocation_line` back to `0` each iteration, so it won't use the value extracted from previous iteration.

Reply:
Good Point!
This also applies to other variables like operation_first.
I initialize all of them at the beginning of the loop.